### PR TITLE
[Kotlin Migration] TranslationActivity

### PR DESCRIPTION
**Pull request by Akshay0701**

### CODE REVIEW

**Comment by Akshay0701** [comment]
forgot to add file name in first commit...

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/TranslationActivity.kt [hunk 1]
```diff
@@ -17,447 +17,353 @@
...
+import com.ichi2.anki.multimediacard.glosbe.json.Response
+import com.ichi2.anki.multimediacard.language.LanguagesListerGlosbe
+import com.ichi2.anki.multimediacard.language.LanguagesListerGlosbe.Companion.requestToResponseLangCode
+import com.ichi2.anki.runtimetools.TaskOperations.stopTaskGracefully
+import com.ichi2.anki.web.HttpFetcher.fetchThroughHttp
+import com.ichi2.async.Connection
+import com.ichi2.libanki.Utils
+import com.ichi2.themes.Themes.disableXiaomiForceDarkMode
+import com.ichi2.ui.FixedTextView
+import com.ichi2.utils.AdaptionUtil.isUserATestClient
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/TranslationActivity.kt:43]: I'd prefer for `AdaptionUtil.isUserATestClient` to be at the call site, so remove this import

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/TranslationActivity.kt [hunk 2]
```diff
@@ -17,447 +17,353 @@
...
-    private ArrayList<String> mPossibleTranslations;
-    private String mLangCodeTo;
-    private BackgroundPost mTranslationLoadPost = null;
-
-
-    private void finishCancel() {
-        Intent resultData = new Intent();
-        setResult(RESULT_CANCELED, resultData);
-        finish();
+open class TranslationActivity : FragmentActivity(), DialogInterface.OnClickListener, DialogInterface.OnCancelListener {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/TranslationActivity.kt:53]: `KotlinCleanup("lateinit")`

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/TranslationActivity.kt [hunk 3]
```diff
@@ -17,447 +17,353 @@
...
-
-        ObjectMapper objectMapper = AnkiSerialization.getObjectMapper();
-        Response resp;
-        try {
-            resp = objectMapper.readValue(mTranslation, Response.class);
-        } catch (JsonProcessingException e) {
-            Timber.w(e);
-            returnFailure(getText(R.string.multimedia_editor_trans_getting_failure).toString());
-            return;
+        val objectMapper = objectMapper
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/TranslationActivity.kt:232]: Original call was cleaner: `AnkiSerialization.getObjectMapper();`

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/TranslationActivity.kt [hunk 4]
```diff
@@ -17,447 +17,353 @@
...
+            return
         }
-
-        mPossibleTranslations = parseJson(resp, mLangCodeTo);
-
-        if (mPossibleTranslations.isEmpty()) {
-            if (!mSource.toLowerCase(Locale.getDefault()).contentEquals(mSource)) {
-                showToastLong(gtxt(R.string.multimedia_editor_word_search_try_lower_case));
+        mPossibleTranslations = parseJson(resp, mLangCodeTo)
+        if (mPossibleTranslations!!.isEmpty()) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/TranslationActivity.kt:252]: could you keep the spacing as it was, quite a bit of this method has changed

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/TranslationActivity.kt [hunk 5]
```diff
@@ -17,447 +17,353 @@
...
-                index = i;
+            /*
+         * The algorithm below includes the parsing of glosbe results. Glosbe.com returns a list of different phrases in
+         * source and destination languages. This is done, probably, to improve the reader's understanding. We leave
+         * here only the translations to the destination language.
+         */
+            val tucs = resp.tuc ?: return ArrayList(0)
+            val res = ArrayList<String>(tucs.size)
+            val desiredLang = requestToResponseLangCode(languageCodeTo!!)
+            for (tuc in tucs) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/TranslationActivity.kt:336]: `KotlinCleanup("mapNotNull")

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
few nits, thanks!!!

**Review by david-allison-1 — APPROVED** [review]
LGTM

**Review by mikehardy — APPROVED** [review]

**Pull request merged**